### ISSUE TITLE
Result of FID on VAE

### ISSUE THREAD

**Issue by shunlinlu** [ChenFengYe/motion-latent-diffusion#9]
Hi, authors

Thank you for presenting such great work. I followed your instructions on Humanml3d and got 0.3 FID on 2200 epochs. I saw that the FID of VAE on your  paper is 0.017, which is pretty low. However, you said the normal range is around 0.45~1.0 in this issue https://github.com/ChenFengYe/motion-latent-diffusion/issues/5#issuecomment-1375037162. 

I'm just wondering what makes such differences here? My config is the same as yours except for the batch size. I'm using batch size 128 with 4 gpus.

Thank you 
Shunlin Lu

**Comment by ChenFengYe** [comment]
Hi Shunlin Lu,

Thanks for your attention. In our paper, the FID of MLD-7 is 0.017 for vae evaluation (Tab. 4) and 1.011 for diffusion evaluation (Tab. 5, text-to-motion). The current config is for MLD-1 (0.247 for vae, 0.474 for diffusion). Your FID (0.3) is quite close to our best evaluation (0.247), and it can be helpful to change batch size and add training epochs.

[The issue 5](https://github.com/ChenFengYe/motion-latent-diffusion/issues/5#issuecomment-1375037162) is not accurate enough. The normal range for diffusion (text-to-motion task) is around [0.45, 1.0]. It should be [0.2, 0.4] for VAE. 

For MLD-1 and MLD-7, MLD-7 only works best in evaluating VAE models (Tab. 4), and MLD-1 wins these generation tasks (Tab. 1, 2, 3, 6). In other words, MLD-7 wins the first training stage for the VAE part, while MLD-1 wins the second for the diffusion part. We thought MLD-7 should perform better than MLD-1 in several tasks, but the results differ. The main reason for this downgrade of a larger latent size, we believe, is the small amount of training data. HumanML3D only includes 15k motion sequences, much smaller than billions of images in image generation. MLD-7 could work better when the motion data amount reaches the million level.

Best wish
Xin

**Comment by shunlinlu** [comment]
Thanks for the clarify. Gonna close this.

**Comment by ChenFengYe** [comment]
Hi, if your VAE results are not correct, please pay attention to this issue #18. We have fixed the bug on KL loss.

https://github.com/ChenFengYe/motion-latent-diffusion/blob/719c219bd059e14f84f2571ab6975855ede0d819/mld/models/losses/mld.py#L105
